op-service/eth/ssz_test.go

```diff
@@ -250,21 +250,21 @@ func TestOPB04(t *testing.T) {
 
 func createPayloadWithWithdrawals(w *types.Withdrawals) *ExecutionPayload {
 	return &ExecutionPayload{
-		ParentHash:    common.HexToHash("0x123"),
-		FeeRecipient:  common.HexToAddress("0x456"),
-		StateRoot:     Bytes32(common.HexToHash("0x789")),
-		ReceiptsRoot:  Bytes32(common.HexToHash("0xabc")),
+		ParentHash:    common.HexToHash("0x1234"),
+		FeeRecipient:  common.HexToAddress("0x4567"),
+		StateRoot:     Bytes32(common.HexToHash("0x6789")),
+		ReceiptsRoot:  Bytes32(common.HexToHash("0xabcd")),
 		LogsBloom:     Bytes256{byte(13), byte(14), byte(15)},
-		PrevRandao:    Bytes32(common.HexToHash("0x111")),
+		PrevRandao:    Bytes32(common.HexToHash("0x1111")),
 		BlockNumber:   Uint64Quantity(222),
 		GasLimit:      Uint64Quantity(333),
 		GasUsed:       Uint64Quantity(444),
 		Timestamp:     Uint64Quantity(555),
-		ExtraData:     common.Hex2Bytes("0x666"),
+		ExtraData:     common.Hex2Bytes("0x6666"),
 		BaseFeePerGas: *uint256.NewInt(777),
-		BlockHash:     common.HexToHash("0x888"),
+		BlockHash:     common.HexToHash("0x8888"),
 		Withdrawals:   w,
-		Transactions:  []Data{common.Hex2Bytes("0x999")},
+		Transactions:  []Data{common.Hex2Bytes("0x9999")},
```

Comment:
Similarly this `Hex2Bytes` call should not have an `0x` prefix.

Reply:
oh, i see it. It is wired this method not support `0x` prefix except `HexToHash` and `HexToAddress`. Should we change `Hex2Bytes` and make the behavior consistent? I also see some other test file not use it correct.

Reply:
`Hex2Bytes` comes from upstream geth so it's not something we'd change like this (we try to minimise the diff). It is a real catch though so it may be worth asking the geth maintainers if they'd accept it as a change.

A semgrep or go-lint rule that at least flagged cases where a constant string with `0x` prefix used like this would make a lot of sense though.